homeassistant/components/kaleidescape/__init__.py

```diff
@@ -0,0 +1,88 @@
+"""The Kaleidescape integration."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+import logging
+from typing import TYPE_CHECKING
+
+from kaleidescape import Device as KaleidescapeDevice, KaleidescapeError
+
+from homeassistant.const import CONF_HOST, EVENT_HOMEASSISTANT_STOP, Platform
+from homeassistant.exceptions import ConfigEntryNotReady, HomeAssistantError
+
+from .const import DOMAIN
+
+if TYPE_CHECKING:
+    from homeassistant.config_entries import ConfigEntry
+    from homeassistant.core import Event, HomeAssistant
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = [Platform.MEDIA_PLAYER]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Kaleidescape from a config entry."""
+    device = KaleidescapeDevice(
+        entry.data[CONF_HOST], timeout=5, reconnect=True, reconnect_delay=5
+    )
+
+    try:
+        await device.connect()
+    except (KaleidescapeError, ConnectionError) as err:
+        await device.disconnect()
+        _LOGGER.error("Unable to connect: %s", err)
+        raise ConfigEntryNotReady from err
```

Comment:
```suggestion
        raise ConfigEntryNotReady(f"Unable to connect to {entry.data[CONF_HOST]}: {err}") from err
```

We don't want retries to flood the log.  Additional retries will automatically downgrade to debug when the message is passed via the exeception.

Reply:
Fixed. Good to know.